boards: pic32-wifire: Fix uart initialization

francois-berder opened the pull request:
### Contribution description

stdio_uart was not working on PIC32 WiFire because `STDIO_UART_DEV` was not defined. This PR also includes some cleanup (remove call to uart_init in board_init, remove useless macros).

### Testing procedure

Compile hello-world and check the output.

### Issues/PRs references

Depends on ~~#11986~~ and ~~#12227~~

benpicco commented:
The changes look good and UART is broken now anyway - please squash.

kaspar030 commented:
This is broken because default for stdio_uart is 115200, but boards/pic32-wifire/Makefile.include sets BAUD ?= 9600. Removing that line makes this work fine with 115200 baud.

I'll push a change here.

kaspar030 commented:
> This is broken because default for stdio_uart is 115200, but boards/pic32-wifire/Makefile.include sets BAUD ?= 9600. Removing that line makes this work fine with 115200 baud.
> 
> I'll push a change here.

Sorry, that got introduced by #9259.

kaspar030 commented:
Please squash, rebase on #9259 (or wait until that is merged), and when resolving the conflicts remove that "BAUD?=9600" line...

Otherwise ACK, I tested that combination.

miri64 commented:
(#9259 was merged)

francois-berder commented:
Rebased and squashed.

> This is broken because default for stdio_uart is 115200, but boards/pic32-wifire/Makefile.include sets BAUD ?= 9600. Removing that line makes this work fine with 115200 baud.

My last commit takes care of that.

kaspar030 commented:
Thanks @francois-berder!
